compositor_pipeline/src/pipeline/pipeline_output.rs

```diff
@@ -1,29 +1,35 @@
 use crate::error::RegisterOutputError;
 
 use super::{
-    encoder::{VideoEncoder, VideoEncoderOptions},
+    encoder::{opus_encoder, AudioEncoderOptions, AudioEncoderPreset, Encoder, EncoderOptions},
     output::Output,
     PipelineOutput, Port, RegisterOutputOptions,
 };
 
 pub(super) fn new_pipeline_output(
     opts: RegisterOutputOptions,
+    output_sample_rate: u32,
 ) -> Result<(PipelineOutput, Option<Port>), RegisterOutputError> {
     let RegisterOutputOptions {
         output_id,
         output_options,
         video,
         audio,
     } = opts;
-    let encoder_opts = video.clone().unwrap().encoder_opts;
-    let VideoEncoderOptions::H264(ref h264_opts) = encoder_opts;
-    if h264_opts.resolution.width % 2 != 0 || h264_opts.resolution.height % 2 != 0 {
-        return Err(RegisterOutputError::UnsupportedResolution(
-            output_id.clone(),
-        ));
-    }
+    let (has_video, has_audio) = (video.is_some(), audio.is_some());
 
-    let (encoder, packets) = VideoEncoder::new(encoder_opts)
+    let encoder_opts = EncoderOptions {
+        video: video.map(|video_opts| video_opts.encoder_opts),
+        audio: audio.map(|audio_opts| {
+            AudioEncoderOptions::Opus(opus_encoder::Options {
+                sample_rate: output_sample_rate,
+                channels: audio_opts.channels,
+                preset: AudioEncoderPreset::Quality,
```

Comment:
pass output_sample_rate to the Encoder::new(), this is not a place for encoder specific code

Also, you seem to be overriding preset from API